### PR TITLE
refactor!: Adjust package managers for external use

### PR DESCRIPTION
**Pull request by nderjung**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

This PR series adjusts the use of the embedded kraftkit package managers for external use as Go packages. No functional changes occur, only API changes.

### CODE REVIEW

File: packmanager/manager.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ import (
 // and output during "construction", particularly providing access to a
 // referencable context with which they can access (within the context of
 // KraftKit) the logging, IOStreams and Config subsystems.
-type NewManagerConstructor func(context.Context) (PackageManager, error)
+type NewManagerConstructor func(context.Context, ...any) (PackageManager, error)
```
Comment (craciunoiuc) [inline packmanager/manager.go:19]: scary :sweat_smile:

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
image


I suppose we'll never know `and` what

**Comment by antoineco** [comment]
> and, adds the ability [...]
:)

**Comment by nderjung** [comment]
@craciunoiuc this is good to go, can you have another look?

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>
Approved-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Pull request merged**